### PR TITLE
[MON-2252] slack app multiple redirect urls fix

### PR DESCRIPTION
**Pull request by yromanyshyn**

### CODE REVIEW

File: backend/deepchecks_monitoring/ee/integrations/slack.py [hunk 1]
```diff
@@ -94,13 +94,18 @@ def generate_authorization_url(self, state, redirect_path) -> str:
...
+        code: str,
+        redirect_uri: t.Optional[str] = None
+    ) -> SlackInstallationSchema:
         """Finish the slack installation."""
         try:
             response = self.client.oauth_v2_access(
                 code=code,
                 client_id=self.settings.slack_client_id,
                 client_secret=self.settings.slack_client_secret.get_secret_value(),
+                redirect_uri=redurect_uri
```
Comment (matanper) [inline backend/deepchecks_monitoring/ee/integrations/slack.py:108]: typo

File: backend/deepchecks_monitoring/ee/api/v1/slack.py [hunk 2]
```diff
@@ -130,7 +130,8 @@ async def installation_callback(
         )
 
     try:
-        installation = utils.finish_installation(code)
+        redirect_path = str(request.url_for('slack-installation-callback'))
```
Comment (matanper) [inline backend/deepchecks_monitoring/ee/api/v1/slack.py:133]: is it safe to use `url_for` even if the server is behind a proxy or something?

**Review by matanper — APPROVED** [review]

File: backend/deepchecks_monitoring/ee/api/v1/slack.py (hunk 2 above)
Comment (matanper) [reply backend/deepchecks_monitoring/ee/api/v1/slack.py:133]: we also use it in eula redirect and authorization redirect, I wonder if we need to replace it in all.
we do already have the `deployment_url` in the settings

Comment (matanper) [reply backend/deepchecks_monitoring/ee/api/v1/slack.py:133]: I checked with local proxy, it works if the proxy adds a "preserver host" header

File: backend/deepchecks_monitoring/ee/integrations/slack.py (hunk 1 above)
Comment (yromanyshyn) [reply backend/deepchecks_monitoring/ee/integrations/slack.py]: fixed

File: backend/deepchecks_monitoring/ee/api/v1/slack.py (hunk 2 above)
Comment (yromanyshyn) [reply backend/deepchecks_monitoring/ee/api/v1/slack.py:133]: cool, thanks!

**Pull request merged**